src/components/core/pivotTable/ResizedColumnsStore.ts

```diff
@@ -93,6 +142,28 @@ export class ResizedColumnsStore {
                 this.manuallyResizedColumns = omit(this.manuallyResizedColumns, colId);
             }
         });
+        this.weakMeasuresColumnWidths = {};
+    }
+
+    public removeWeakMeasureColumn(column: Column) {
+        const weakColumnWidth = this.getMatchedWeakMeasuresColumnWidths(column);
+        if (weakColumnWidth) {
+            this.weakMeasuresColumnWidths = omit(
+                this.weakMeasuresColumnWidths,
+                weakColumnWidth.measureColumnWidthItem.locator.measureLocatorItem.measureIdentifier,
+            );
+            const colIds = Object.keys(this.manuallyResizedColumns);
+            colIds.forEach(colId => {
+                const item = this.manuallyResizedColumns[colId];
+                if (
+                    isColumnWidthAuto(item.width) &&
+                    this.isMatchingWeakWidth(item, weakColumnWidth) &&
+                    !this.isAllMeasureColumWidthUsed()
+                ) {
+                    this.manuallyResizedColumns = omit(this.manuallyResizedColumns, colId);
```

Comment:
It's just a minor thing but what about avoiding forEach and using https://lodash.com/docs/#omitBy ? Something like (not tested)
```
this.manuallyResizedColumns = omitBy(this.manuallyResizedColumns,  item => 
    isColumnWidthAuto(item.width) &&
    this.isMatchingWeakWidth(item, weakColumnWidth) &&
    !this.isAllMeasureColumWidthUsed()
);
```

Reply:
done